app/views/documents/show/_history_tab.html.erb

```diff
@@ -1,31 +1,39 @@
 <% displayed_edition_number = nil %>
 
-<% @document.timeline_entries.includes(:user).order(created_at: :desc).includes(:retirement).includes(:removal).each do |entry| %>
-  <div class="app-timeline-entry govuk-!-margin-bottom-6">
-    <% if displayed_edition_number != entry.edition_number %>
-      <h3 class="govuk-heading-m"><%= t "documents.history.edition_header", number: entry.edition_number.ordinalize %></h3>
-      <% displayed_edition_number = entry.edition_number %>
-    <% end %>
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <% @document.timeline_entries.includes(:user).order(created_at: :desc).includes(:retirement, :removal, :internal_note).each do |entry| %>
+      <div class="app-timeline-entry govuk-!-margin-bottom-6">
+        <% if displayed_edition_number != entry.edition_number %>
+          <h3 class="govuk-heading-m"><%= t "documents.history.edition_header", number: entry.edition_number.ordinalize %></h3>
+          <% displayed_edition_number = entry.edition_number %>
+        <% end %>
 
-    <h4 class="govuk-heading-s govuk-!-margin-bottom-1">
-      <%= t "documents.history.entry_types.#{entry.entry_type}" %>
-    </h4>
+        <h4 class="govuk-heading-s govuk-!-margin-bottom-1">
+          <%= t "documents.history.entry_types.#{entry.entry_type}" %>
+        </h4>
 
-    <% if entry.retirement.present? %>
-      <%= entry.retirement.explanatory_note %>
-    <% elsif entry.removal.present? %>
-      <% if entry.removal.explanatory_note.present? %>
-        <p><%= entry.removal.explanatory_note %></p>
-      <% end %>
-      <% if entry.removal.redirect %>
-        <p>Redirected to <%= link_to(Plek.new.website_root + entry.removal.alternative_path) %></p>
-      <% elsif entry.removal.alternative_path.present? %>
-        <p><%= link_to(Plek.new.website_root + entry.removal.alternative_path) %></p>
-      <% end %>
-    <% end %>
+        <div class="app-timeline-entry__dateline">
+          <%= entry.created_at.strftime("%d %B %Y at %l:%M%P") %> by <%= entry.username_or_unknown %>
+        </div>
 
-    <div class="app-timeline-entry__dateline">
-      <%= entry.created_at.strftime("%d %B %Y at %l:%M%P") %> by <%= entry.username_or_unknown %>
-    </div>
+        <% if entry.internal_note.present? %>
+          <%= simple_format(Rinku.auto_link(entry.internal_note.body)) %>
```

Comment:
So what I think this should do is:

- take the note body and convert HTML into HTML entities
- then run auto link
- then run simple format

That would mean if a user entered any HTML it would be preserved.

So in terms of test cases:

input: "here is a note\nwith new lines" should output "<p>here is a note\n<br />with new lines</p>"(which it does currently)
input: "here is a note\nwith new lines\nand an email@example.com address" should output: "<p>here is a note\n<br />with new lines\n<br />and an <a href=\"mailto:email@example.com\">email@example.com</a> address</p>" (which it does currently)
input: "here is a note with <b>html</b> inside it" should output: "<p>here is a note with &lt;b&gt;html&lt;/b&gt; inside it</p>" (currently renders the bold)
input: "here is a note with a <script>script</script> inside it" should output: "<p>here is a note with &lt;script&gt;script&lt;/script&gt; inside it</p>" (currently strips the tag)

Reply:
Thanks for the detailed explanation! I've made the changes as you've requested so the html is escaped and links / line breaks are added and have tested accordingly.